File: challenges/medium/05_inventory_analyzer/processor.py
def filter_active(items):
    """Return only the items where active=True."""
    return (item for item in items if item["active"])  # something subtle here


def summarize(values):
    """
    Given a list of numeric values, return basic statistics.
    Returns {"count": n, "total": sum, "average": avg}.
    If values is empty, return {"count": 0, "total": 0, "average": 0}.
    """
    pass  # TODO: implement

File: challenges/medium/17_inventory_analyzer_ts/processor.ts
export function* filterActive(items: any[]): Generator<any> {    // bug: generator
  for (const item of items) {
    if (item.active) yield item;
  }
}

export function summarize(values: number[]): { count: number; total: number; average: number } | undefined {
  // TODO: return { count, total, average } — for empty input return all zeros
}
